tests/parsed_data_out_03.cc

```diff
@@ -0,0 +1,126 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2014 by the deal.II authors
+//
+//    This file is subject to LGPL and may not be distributed
+//    without copyright and license information. Please refer
+//    to the file deal.II/doc/license.html for the  text  and
+//    further information on this license.
+//
+//-----------------------------------------------------------
+
+
+#include "tests.h"
+#include "utilities.h"
+#include "parameter_acceptor.h"
+#include "parsed_finite_element.h"
+#include "parsed_grid_generator.h"
+#include "parsed_data_out.h"
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/grid/grid_generator.h>
+
+#include <deal.II/lac/block_vector.h>
+
+#include <deal.II/base/mpi.h>
+
+template<int dim>
+class Test : public ParameterAcceptor
+{
+public:
+  Test();
+  virtual void declare_parameters(ParameterHandler &prm);
+  void run();
+
+private:
+
+  void make_grid_fe();
+  void setup_dofs();
+  void output_results();
+
+  ParsedFiniteElement<dim> fe_builder;
+  ParsedGridGenerator<dim,dim> tria_builder;
+  unsigned int initial_refinement;
+  shared_ptr<Triangulation<dim> >      triangulation;
+  shared_ptr<FiniteElement<dim,dim> >  fe;
+  shared_ptr<DoFHandler<dim> >         dof_handler;
+  BlockVector<double> solution;
+  ParsedDataOut<dim, dim>                 data_out;
+};
+
+template <int dim>
+Test<dim>::Test ()
+  :
+  ParameterAcceptor("Global parameters"),
+  tria_builder("Triangulation"),
+  fe_builder("FE_Q",
+             "FESystem[FE_Q(2)^dim-FE_Q(1)]","u,u,p"),
+  data_out("Data out","vtk","","output")
+{}
+
+template <int dim>
+void Test<dim>::declare_parameters(ParameterHandler &prm)
+{
+  add_parameter(prm, &initial_refinement, "Initial global refinement", "3",
+                Patterns::Integer(0));
+}
+
+template <int dim>
+void Test<dim>::make_grid_fe ()
+{
+  ParameterAcceptor::initialize("parameters_ser.prm", "used_parameters_ser.prm");
+  triangulation = SP(tria_builder.serial());
+  triangulation->refine_global(initial_refinement);
+  dof_handler = SP(new DoFHandler<dim>(*triangulation));
+  fe=SP(fe_builder());
+}
+
+template <int dim>
+void Test<dim>::setup_dofs ()
+{
+  std::vector<unsigned int> sub_blocks (dim+1,0);
+  sub_blocks[dim] = 1;
+  dof_handler->distribute_dofs (*fe);
+
```

Comment:
Al momento la cosa fa quello che ti aspetti per caso, perché non hai riordinato i gradi di libertà per blocchi. So che stai sputando fuori solo vettori nulli, ma se tu provassi a settare ad uno il blocco delle velocità, in uscita vedresti gli uni sparsi in giro un po' ovunque, anche nelle pressioni, perché l'ordinamento attuale del dh e' `u1[0] u2[0] p[0] u1[1] u2[1] p[1] ...` e non, come è nelle tue intenzioni, `u1[0] u2[0] u1[1] u2[1] .... p[0] p[1] ...`